ozaria/site/components/interactive/PageInteractive/index.vue

```diff
@@ -30,16 +32,52 @@
       }
     },
 
-    data: () => ({
-      loading: false,
-      interactive: {},
-      interactiveSession: {},
-      interactiveType: ''
-    }),
-
     computed: {
+      ...mapGetters('interactives', [
+        'currentInteractiveDataLoading',
+        'currentInteractive',
+        'currentInteractiveSession'
+      ]),
+
+      interactiveConfig () {
+        switch (this.currentInteractive.interactiveType) {
+        case 'draggable-statement-completion':
+          return this.currentInteractive.draggableStatementCompletionData
+
+        case 'insert-code':
+          return this.currentInteractive.insertCodeData
+
+        case 'draggable-ordering':
+          return this.currentInteractive.draggableOrderingData
+
+        default:
+          noty({ text: 'Interactive type is not set for the interactive', type: 'error', timeout: '2000' })
+          throw new Error('Invalid interactive type provided for interactive')
+        }
+      },
+
+      localizedInteractiveConfig () {
```

Comment:
This is a reimplementation of https://github.com/codecombat/codecombat/blob/ff52499aaf339466a528db3500d3efc6309d7f93/app/core/utils.coffee#L226

This implementation supports nested arrays (not supported by the current implementation) and generates the internationalized config once instead of on each call to the method.

Would be good to get an additional review of the logic on this.  Also may make sense to put this in a utility method for use elsewhere.  Thoughts?

Reply:
Yep this can be moved to `ozariaUtils.js`.